03-react/06-form/src/main.tsx

```diff
@@ -4,9 +4,92 @@ import * as ReactDOM from "react-dom";
 const appDiv = document.getElementById("app");
 
 if (!(appDiv instanceof HTMLDivElement)) {
-	throw new Error("No div with id 'app' found");
+  throw new Error("No div with id 'app' found");
 }
-
-const App = (): JSX.Element => {return<></>};
+interface UserForm {
+  firstname: string;
+  lastname: string;
+  age: string;
+}
+interface Fromprops {
+  onSumbit: (x: UserForm) => void;
+  onChange: (x: UserForm) => void;
+}
+const Form = (props: Fromprops): JSX.Element => {
+  const [input, setInput] = React.useState<UserForm>({
+    firstname: "",
+    lastname: "",
+    age: "",
+  });
+  return (
+    <>
+      <form
+        action="#"
+        method="post"
+        onSubmit={(e) => {
+          e.preventDefault();
+          props.onSumbit(input);
+        }}
+      >
+        <label htmlFor="firstname">First Name:</label>
+        <input
+          type="text"
+          id="firstname"
+          value={input.firstname}
+          onChange={(e) => {
+            setInput({ ...input, firstname: e.target.value });
+            props.onChange({
+              firstname: "",
+              lastname: "",
+              age: "",
```

Comment:
So you only call `props.onChange` with an object containing empty strings? Why?

Reply:
I only want to show the input when submit, and empty it when change the input.